tests/mir-opt/reference_prop.rs

```diff
@@ -0,0 +1,428 @@
+// unit-test: ReferencePropagation
+
+#![feature(raw_ref_op)]
+#![feature(core_intrinsics, custom_mir)]
+
+#[inline(never)]
+fn opaque(_: impl Sized) {}
+
+fn reference_propagation<'a, T: Copy>(single: &'a T, mut multiple: &'a T) {
+    // Propagation through a reference.
+    {
+        let a = 5_usize;
```

Comment:
The mir files for these tests are pretty hard to read right now. Can you put an `opaque(())` at the end of all the test "sections"? That way we these get divided up into basic blocks which seems like it'd be much easier